CMakeLists.txt

```diff
@@ -124,7 +133,7 @@ if (CMAKE_CXX_COMPILER_ID MATCHES "(Clang|GNU)")
     endif ()
 
     if (ADDRESS_SANITIZER)
-        add_compile_options(-fsanitize=address -fsanitize-address-use-after-scope -fsanitize=leak -fno-omit-frame-pointer -fstack-protector-strong -fstack-clash-protection) # additional flags: -D_GLIBCXX_DEBUG -D_FORTIFY_SOURCE=2
+        add_compile_options(-fsanitize=address -fsanitize-address-use-after-scope -D_GLIBCXX_SANITIZE_VECTOR=1 -fsanitize=leak -fno-omit-frame-pointer -fstack-protector-strong -fstack-clash-protection) # additional flags: -D_GLIBCXX_DEBUG -D_FORTIFY_SOURCE=2
```

Comment:
Similar to the comment above...

Reply:
Really? I looked at the code, and obviously it doesn't come for free, but it doesn't look heavy to me. asan itself is waay heavier. And this just makes asan more precise when dealing with `std::vector`.